optuna/_callbacks.py

```diff
@@ -136,3 +131,21 @@ def retried_trial_number(trial: FrozenTrial) -> Optional[int]:
         """
 
         return trial.system_attrs.get("failed_trial", None)
+
+    @staticmethod
+    @experimental("3.0.0")
+    def retry_history(trial: FrozenTrial) -> List[int]:
+        """Return the list of retried trial IDs with respect to the specified trial.
```

Comment:
Trial id and trial number are different conceptions in Optuna.

```suggestion
        """Return the list of retried trial numbers with respect to the specified trial.
```